iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -625,13 +625,13 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
 
                 _handleDisconnectCts.Cancel();
                 await base.CloseAsync(cancellationToken).ConfigureAwait(false);
-                Dispose(true);
```

Comment:
Since in the finally statement we are releasing the semaphore, we need to dispose the object after the semaphore has been released to avoid nullref.